src/App.tsx

```diff
@@ -15,37 +15,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT = 'DEFAULT',
+  ALPHABETICAL = 'ALPHABETICAL',
+  LENGTH = 'LENGTH',
+}
 export const App: React.FC = () => {
+  const initialGoods = [...goodsFromServer];
+  const [goods, setGoods] = useState<string[]>(initialGoods);
+  const [sortType, setSortType] = useState<SortType>(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortGoods = (type: SortType) => {
+    let sortedGoods = [...initialGoods];
+
+    switch (type) {
+      case SortType.ALPHABETICAL:
+        sortedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SortType.LENGTH:
+        sortedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        sortedGoods = initialGoods;
+        break;
+    }
+
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
+
+    setGoods(sortedGoods);
+    setSortType(type);
+  };
+
+  const toggleReverse = () => {
+    setGoods(prevGoods => [...prevGoods].reverse());
+    setIsReversed(!isReversed);
+  };
+
+  const resetGoods = () => {
+    setGoods([...initialGoods]);
+    setSortType(SortType.DEFAULT);
+    setIsReversed(false);
+  };
+
+  const isInitialOrder = JSON.stringify(goods) === JSON.stringify(initialGoods);
```

Comment:
The `isInitialOrder` check using `JSON.stringify` might not work as expected if the order of elements changes but the elements themselves remain the same. Consider using a different method to compare the order of elements, such as comparing each element in the array directly.